conda/gateways/disk/create.py

```diff
@@ -315,28 +315,13 @@ def create_package_cache_directory(pkgs_dir):
     # returns False if package cache directory cannot be created
     try:
         log.trace("creating package cache directory '%s'", pkgs_dir)
-        mkdir_p(pkgs_dir)
-        touch(join(pkgs_dir, 'urls'))
-        touch(join(pkgs_dir, PACKAGE_CACHE_MAGIC_FILE))
+        sudo_safe = expand(pkgs_dir).startswith(expand('~'))
```

Comment:
What if a user has defined a package cache directory using `pkgs_dirs` in ~/.condarc to some place other than his `~`? Could be a shared location among users of the same primary group.

Reply:
Not ideal, but this is all current behavior, even in 4.3. The config for `pkgs_dirs` and `envs_dirs` *is* independent though. What’ll happen if a user defines a custom envs dir is that, to test permissions, a `pkgs` directory will be created along side it, then an empty `urls.txt` file in that directory. Like I said, not ideal. But itshow conda 4.3 does it, so the behavior the user sees at least doesn’t change.

Reply:
No, I meant `sudo_safe` would be false, if the `pkgs_dir` is a custom path.

Reply:
I’m not sure right now how to improve on it. I wanted to add the sudo_safe stuff to to help people who shouldn’t be using sudo at all. If people are sophisticated enough to be customizing directories like that, then for now I’m ok hoping they’re also sophisticated enough to dig themselves out of a sudo hole if they manage to get into one.

Reply:
Fair enough.